dev/src/util.ts

```diff
@@ -75,6 +77,23 @@ export function isObject(value: unknown): value is {[k: string]: unknown} {
   return Object.prototype.toString.call(value) === '[object Object]';
 }
 
+/**
+ * Verifies that 'obj' is a plain JavaScript object that can be encoded as a
+ * 'Map' in Firestore.
+ *
+ * @private
+ * @param input The argument to verify.
+ * @returns 'true' if the input can be a treated as a plain object.
+ */
+export function isPlainObject(input: unknown): input is DocumentData {
+  return (
+    isObject(input) &&
+    (Object.getPrototypeOf(input) === Object.prototype ||
+      Object.getPrototypeOf(input) === null ||
+      input.constructor.name === 'Object')
```

Comment:
According to this stackoverflow [post](https://stackoverflow.com/questions/7039260/can-i-use-constructor-name-to-detect-types-in-javascript), `obj.constructor` can be changed by the user, so they recommend using `instanceof` instead.

Reply:
The recommendation here does not quite work for us. We don't have access to the user's type and `instanceof Object` returns true for custom types as well as plain objects such as `{}`.

It also looks like the other suggestion `Object.prototype.toString.call` just returns `[object Object]` for custom types, so that is likely out of the question too,